Snowflake.Data.Tests/IntegrationTests/SFDbCommandIT.cs

```diff
@@ -175,6 +175,289 @@ public void TestExecuteAsyncWithMaxRetryReached()
                 Assert.GreaterOrEqual(stopwatch.ElapsedMilliseconds, 30 * 1000);
             }
         }
+
+        [Test]
+        public async Task TestAsyncExecQueryAsync()
+        {
+            string queryId;
+            var expectedWaitTime = 5;
+
+            using (SnowflakeDbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString;
+                await conn.OpenAsync(CancellationToken.None).ConfigureAwait(false);
+
+                using (SnowflakeDbCommand cmd = (SnowflakeDbCommand)conn.CreateCommand())
+                {
+                    // Arrange
+                    cmd.CommandText = $"CALL SYSTEM$WAIT({expectedWaitTime}, \'SECONDS\');";
+
+                    // Act
+                    queryId = await cmd.ExecuteAsyncInAsyncMode(CancellationToken.None).ConfigureAwait(false);
+                    var queryStatus = await cmd.GetQueryStatusAsync(queryId, CancellationToken.None).ConfigureAwait(false);
+
+                    // Assert
+                    Assert.IsTrue(QueryStatuses.IsStillRunning(queryStatus));
+
+                    // Act
+                    DbDataReader reader = await cmd.GetResultsFromQueryIdAsync(queryId, CancellationToken.None).ConfigureAwait(false);
+                    queryStatus = await cmd.GetQueryStatusAsync(queryId, CancellationToken.None).ConfigureAwait(false);
+
+                    // Assert
+                    Assert.IsTrue(reader.Read());
+                    Assert.AreEqual($"waited {expectedWaitTime} seconds", reader.GetString(0));
+                    Assert.AreEqual(QueryStatus.Success, queryStatus);
+                }
+
+                await conn.CloseAsync(CancellationToken.None).ConfigureAwait(false);
+            }
+        }
+
+        [Test]
+        public async Task TestMixedSyncAndAsyncQueryAsync()
```

Comment:
maybe let's think about the better name for the test or changing what we test.
When I read  the name I thought that we test query executions in async mode and then we would demonstrate that  GetResultsFromQueryId(Async) - for both versions work properly. But in the test only GetResultsFromQueryIdAsync and GetQueryStatusAsync are tested and GetResultsFromQueryId and GetQueryStatus are not tested.

Reply:
I understand the test name can be confusing, it's now renamed to `TestExecuteNormalQueryWhileAsyncExecQueryIsRunning`

The purpose of the test was to show that it's possible to run a normal query while an async exec query is running